lib/OpenQA/Setting.pm

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2017-2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Setting;
+
+use warnings;
+use strict;
+
+sub new {
+    my ($class, %args) = @_;
+    my $self = {};
+
+    for my $key (keys %args) {
+        $self->{$key} = $args{$key};
+    }
+    bless $self, $class;
+    return $self;
+}
+
+sub replace_setting {
+    my ($self) = @_;
+
+    for my $value (values %{$self}) {
```

Comment:
I think this deserves a comment, e.g. `# replace %NAME% in values with $self->{NAME}`.